metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanSystemAspect.java

```diff
@@ -129,6 +164,64 @@ public EbeanSystemAspect forUpdate(
       this.ebeanAspectV2 = ebeanAspectV2;
       this.urn = UrnUtils.getUrn(ebeanAspectV2.getUrn());
       this.aspectName = ebeanAspectV2.getAspect();
+
+      // Pre-patch validation: check existing aspect size from DB
+      if (prePatchValidationConfig != null
+          && prePatchValidationConfig.getPrePatch() != null
+          && prePatchValidationConfig.getPrePatch().isEnabled()) {
+        String rawMetadata = ebeanAspectV2.getMetadata();
+        if (rawMetadata != null) {
+          long actualSize = rawMetadata.length();
+          long threshold = prePatchValidationConfig.getPrePatch().getMaxSizeBytes();
+
+          if (actualSize > threshold) {
+            OversizedAspectRemediation remediation =
+                prePatchValidationConfig.getPrePatch().getOversizedRemediation();
+
+            log.warn(
+                "Oversized pre-patch aspect remediation={}: urn={}, aspect={}, size={} serialized bytes, threshold={} serialized bytes",
+                remediation.logLabel,
+                urn,
+                aspectName,
+                actualSize,
+                threshold);
+
+            // Handle oversized aspect according to remediation strategy
+            if (remediation == OversizedAspectRemediation.REPLACE_WITH_PATCH && aspectDao != null) {
+              try {
+                aspectDao.deleteAspect(urn, aspectName, 0L);
```

Comment:
This isn't the proper way to delete an aspect. In order to process the aspect delete it must go back through the entity service. You'll want to restructure the logic to avoid the write (leave the SQL row), collect the deletes, and then execute those deletes outside the current transaction.

Reply:
This leaves Elasticsearch indices, consumer hooks, and potentially graph edges unhandled leading to an inconsistent state.